### PR TITLE
DM-37847: add site environments

### PR DESCRIPTION
**Pull request by wguanicedew**

### CODE REVIEW

File: src/admin/site_environments.rst [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+- Realtime logging:
+
+  Environments required for realtime logging::
+
+    # set USE_REALTIME_LOGGING to no if no realtime logging.
+    export REALTIME_LOGGING_SERVER=google-cloud-logging
+    export REALTIME_LOGNAME=Panda-RubinLog
+    export REALTIME_LOGFILES=payload-log.json
+    export USE_REALTIME_LOGGING=yes
```
Comment (MichelleGower) [inline src/admin/site_environments.rst]: Just for readability, I would move the USE_REALTIME_LOGGING export line up right under the comment about setting it to no.   If no realtime logging, should one comment out those other REALTIME env vars or set them to dummy values or ...?

File: src/admin/site_environments.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+- voms and x509 certificates:
+
+  Setup voms-proxy-info and x509 CA certificates (/etc/grid-security/certificates by default).
+  Setup osg for SLAC here (or setup lcg clients here for other sites).
```
Comment (MichelleGower) [inline src/admin/site_environments.rst]: I'm not sure what "here" means in this sentence?  Where they supposed to be links to follow?

File: src/admin/site_environments.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  For EU sites, some lcg libraries are already installed locally. So this line can be ignored::
```
Comment (MichelleGower) [inline src/admin/site_environments.rst]: Would the following be better?   can be ignored => can be commented out

File: src/admin/site_environments.rst [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+    # source /cvmfs/oasis.opensciencegrid.org/mis/osg-wn-client/current/el8-x86_64/setup.sh
+
+- group permission for new directories:
+
+  It’s used for setting the permissions of some sub-directories in the job::
```
Comment (MichelleGower) [inline src/admin/site_environments.rst]: I would include a short sentence why it is required andy why 007.  For example something like, the PanDA jobs will run as a service account and needs to ?   Also is it required that the service account be a special group (e.g., same as particular other accounts)?  The documentation of the service account requirements doesn't go in this section, but I'm not sure if/where it's documented.

File: src/admin/site_environments.rst [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+    umask 007
+
+- lsst accounts:
```
Comment (MichelleGower) [inline src/admin/site_environments.rst]: Would it be better to call this section Authentication?

File: src/admin/site_environments.rst [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+  Accounts to access google realtime logging service, required by pilots to send realtime logs::
+
+    export GOOGLE_APPLICATION_CREDENTIALS=/sdf/home/l/lsstsvc1/.lsst/gcs-access.json
+
+  Butler db accounts::
+
+    export LSST_DB_AUTH=/sdf/home/l/lsstsvc1/.lsst/db-auth.yaml
+
+  Account to access SLAC objectstore, it’s not required for other sites if there are no objectstores::
```
Comment (MichelleGower) [inline src/admin/site_environments.rst]: I'm not 100% sure I followed this.  Here's what I think is meant:
**AWS_SHARED_CREDENTIALS_FILE**:  Authentication for file storage at the site if required, for example an objectstore.  
SLAC example:
export AWS_SHARED_CREDENTIALS_FILE=/sdf/home/l/lsstsvc1/.lsst/aws-credentials.ini

Note:  This is for storage local to the site.  This file does not need to contain storage credentials for all sites.

File: src/admin/site_environments.rst [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    # export AWS_SHARED_CREDENTIALS_FILE=/sdf/home/l/lsstsvc1/.lsst/aws-credentials.ini
+
+- temp directory:
+
+  SLAC temp directory, it’s not required for other sites if the default /tmp works ok
+  (The default /tmp directory should be big enough because PanDA and butler will put
+  some temp files in the /tmp directory. It’s good to make sure that 2~4 GB per CPU cores)::
+
+    export TMPDIR=/lscratch/lsstsvc1/pilot_temp/
+    mkdir -p $TMPDIR
```
Comment (MichelleGower) [inline src/admin/site_environments.rst:84]: Again just some wording suggestions:
**TMPDIR**: The temporary used by PANDA and job (LSST butler) for temp files.  It is recommended that there is at least 2~4 GB per CPU core.  Will default to /tmp if not set.
SLAC example:
```
    export TMPDIR=/lscratch/lsstsvc1/pilot_temp/
￼    mkdir -p $TMPDIR
```

File: src/admin/site_environments.rst [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+
+- butler configuration:
+
+  The location of the butler configuration files.
```
Comment (MichelleGower) [inline src/admin/site_environments.rst]: "butler configuration" might be confused with butler.yaml.  To be clearer, I would say the Butler Repository Index file.

File: src/admin/site_environments.rst [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+  It can be a posix file or s3 objectstore file::
+
+    # export DAF_BUTLER_REPOSITORY_INDEX="s3://butler-us-central1-panda-dev/dc2/butler-external.yaml"
+    # export RUBIN_RUN_TEMP_SPACE="s3://butler-us-central1-panda-dev/"
```
Comment (MichelleGower) [inline src/admin/site_environments.rst]: I don't think you meant to have RUBIN_RUN_TEMP_SPACE here, but meant to have it in the next section.

File: src/admin/site_environments.rst [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+    export DAF_BUTLER_REPOSITORY_INDEX="/sdf/group/rubin/shared/data-repos.yaml"
+
+- LSST_RUN_TEMP_SPACE:
+
+  A temporary directory to store qgraph files, the execution butler and a few other scripts.
+  During 'bps submit', the qgraph files, the execution butler (a sqlite file) and a few other
+  scripts will be created and stored in this directory. These files will be accessed by pipeline jobs.
+  It should be a share directory or a Grid/Cloud directory which can be accessed by all worker nodes::
```
Comment (MichelleGower) [inline src/admin/site_environments.rst:103]: share -> shared

Comment (MichelleGower) [reply src/admin/site_environments.rst:103]: And is it LSST_RUN_TEMP_SPACE or RUBIN_RUN_TEMP_SPACE?  I see both in this document.

File: src/admin/site_environments.rst [hunk 11]
```diff
@@ -0,0 +1,146 @@
+Site Environments
+=================
+
+The site environments required for running PanDA jobs.
```
Comment (MichelleGower) [inline src/admin/site_environments.rst]: Wording suggestion:  At many sites multiple experiments share the computing resource.  Thus experiment specific settings should not be set globally.  Instead, these settings can be put in a file that will be sourced by the PanDA pilot as well as the user.  For an complete example of the file, see section X-link-here.  The next section will describe the content of the file.

File: src/admin/site_environments.rst [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+
+Export LSST_LOCAL_PROLOG
+------------------------
+
+For running PanDA jobs, the site only needs to export LSST_LOCAL_PROLOG.
```
Comment (MichelleGower) [inline src/admin/site_environments.rst:16]: Just to clarify, this is a global export by the site?  Or is this a variable set in the user's bashrc for their interactive use and inside the bps submit yaml to tell the pilot to source it (runnerCommand)?

File: src/admin/site_environments.rst [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+  export LSST_LOCAL_PROLOG=/sdf/home/l/lsstsvc1/.lsst/local_prolog.sh
+
+The content of LSST_LOCAL_PROLOG
+--------------------------------
+
+Here are the content of LSST_LOCAL_PROLOG and corresponding descriptions.
+
+- Timezone:
+
+  To set the timezone, it's good for logging format for jobs from different sites::
```
Comment (MichelleGower) [inline src/admin/site_environments.rst]: Wording:
Set the timezone to ensure that it is consistent in logs for jobs run across sites in multiple timezones.

File: src/admin/site_environments.rst [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+
+    export TZ=UTC
+
+- Realtime logging:
```
Comment (MichelleGower) [inline src/admin/site_environments.rst:34]: Don't know if need a little intro or not.  For example:
To allow operators visibility into running jobs, logs can be sent to a logging service in real time, for example the Google Cloud logging.  To enable this, multiple environment variables need to be set.

File: src/admin/site_environments.rst [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+
+    # set USE_REALTIME_LOGGING to no if no realtime logging.
+    export REALTIME_LOGGING_SERVER=google-cloud-logging
+    export REALTIME_LOGNAME=Panda-RubinLog
+    export REALTIME_LOGFILES=payload-log.json
+    export USE_REALTIME_LOGGING=yes
+
+- voms and x509 certificates:
+
+  Setup voms-proxy-info and x509 CA certificates (/etc/grid-security/certificates by default).
```
Comment (MichelleGower) [inline src/admin/site_environments.rst:48]: What are these usually used for? job submission?

File: src/index.rst [hunk 16]
```diff
@@ -1,9 +1,13 @@
+
 This guide describes basic operations for running Rubin workflow with
 PanDA on the Google Cloud deployment prepared for the DP0.2 exercise.
 This setup continuously evolves, this is why the current manual may not
 always precisely reflect the transitional state. In case of founding any
```
Comment (MichelleGower) [inline src/index.rst]: If any discrepancy is found, please...

**Review by MichelleGower — APPROVED** [review]
I don't know what versions are required, so skipped reviewing the requirements.txt.  I have not read all the admin documentation so if something is documented elsewhere that I mention in my comments maybe consider adding a link here.  Many suggestions are just wording suggestions (mostly moving to the beginning the why to lead the reader into the what).  And a couple were places I didn't know enough and needed clarification.

File: src/admin/site_environments.rst (hunk 2 above)
Comment (wyang007) [reply src/admin/site_environments.rst]: the last comment line (+40): maybe we can say:
For USDF, use OSG clients for voms and x509 tools and infrastructure.

File: src/admin/site_environments.rst (hunk 3 above)
Comment (wyang007) [reply src/admin/site_environments.rst]: Is this (line +41) true for both FrDF and UKDF ?

File: src/admin/site_environments.rst (hunk 6 above)
Comment (wyang007) [reply src/admin/site_environments.rst]: Is this to access butler at SLAC ?

File: src/admin/site_environments.rst (hunk 3 above)
Comment (wguanicedew) [reply src/admin/site_environments.rst]: yes. it's true for FrDF and UKDF.

File: src/admin/site_environments.rst (hunk 6 above)
Comment (wguanicedew) [reply src/admin/site_environments.rst]: yes. I moved it to the butler section and added more comments.

File: src/admin/site_environments.rst (hunk 10 above)
Comment (wguanicedew) [reply src/admin/site_environments.rst:103]: it's LSST_RUN_TEMP_SPACE. I fixed it.

File: src/admin/site_environments.rst (hunk 12 above)
Comment (wguanicedew) [reply src/admin/site_environments.rst:16]: This one is a global export by the site. Pilot wrapper will source it. I modified the instructions.

File: src/admin/site_environments.rst (hunk 15 above)
Comment (wguanicedew) [reply src/admin/site_environments.rst:48]: it's used for pilot authentication.

**Pull request merged**